### PR TITLE
Add description about user ATTRIBUTE

### PR DESCRIPTION
**Pull request by CbcWestwolf**
### What is changed, added or deleted? (Required)

1. Support `ATTRIBUTE` and `COMMENT` in `CREATE USER` and `ALTER USER` statements
2. Add a new column `user_attributes` to `mysql.user`, and add a new view `information_schema.user_attributes`

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.3 (TiDB 6.3 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/pingcap/tidb/pull/38201

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: sql-statements/sql-statement-show-columns-from.md [hunk 1]
```diff
@@ -165,44 +165,49 @@ show full columns from mysql.user;
...
-+-----------------------+---------------+-------------+------+------+---------+-------+---------------------------------+---------+
-33 rows in set (0.01 sec)
++------------------------+---------------+-------------+------+-----+---------+-------+---------------------------------+---------+
+| Field                  | Type          | Collation   | Null | Key | Default | Extra | Privileges                      | Comment |
++------------------------+---------------+-------------+------+-----+---------+-------+---------------------------------+---------+
+| Host                   | char(255)     | utf8mb4_bin | NO   | PRI | <null>  |       | select,insert,update,references |         |
+| User                   | char(32)      | utf8mb4_bin | NO   | PRI | <null>  |       | select,insert,update,references |         |
+| authentication_string  | text          | utf8mb4_bin | YES  |     | <null>  |       | select,insert,update,references |         |
+| plugin                 | char(64)      | utf8mb4_bin | YES  |     | <null>  |       | select,insert,update,references |         |
```
Comment (xiongjiwei) [inline sql-statements/sql-statement-show-columns-from.md:174]: <null> should be `NULL`?

**Review by xiongjiwei — APPROVED** [review]

File: information-schema/information-schema-user-attributes.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
+---
+title: USER_ATTRIBUTES
+summary: 了解 information_schema 表 `USER_ATTRIBUTES`。
```
Comment (Oreoxmt) [inline information-schema/information-schema-user-attributes.md:3]:
```suggestion
summary: 了解 INFORMATION_SCHEMA 表 `USER_ATTRIBUTES`。
```

File: information-schema/information-schema-user-attributes.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+| root      | %    | NULL                                              |
+| testuser1 | %    | {"comment": "This user is created only for test"} |
+| testuser2 | %    | {"email": "user@pingcap.com"}                     |
++-----------+------+---------------------------------------------------+
+3 rows in set (0.00 sec)
+```
+
+`USER_ATTRIBUTES` 表中列的含义如下：
+
+* `USER`：账户的用户名。
```
Comment (Oreoxmt) [inline information-schema/information-schema-user-attributes.md:49]:
```suggestion
* `USER`：用户名。
```

File: information-schema/information-schema-user-attributes.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+| testuser2 | %    | {"email": "user@pingcap.com"}                     |
++-----------+------+---------------------------------------------------+
+3 rows in set (0.00 sec)
+```
+
+`USER_ATTRIBUTES` 表中列的含义如下：
+
+* `USER`：用户名。
+* `HOST`：用户的主机名。
+* `ATTRIBUTE`：用户注释和属性。
```
Comment (Oreoxmt) [inline information-schema/information-schema-user-attributes.md:51]:
```suggestion
* `ATTRIBUTE`：与当前用户相关的注释和属性。
```

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

File: information-schema/information-schema-user-attributes.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+| testuser2 | %    | {"email": "user@pingcap.com"}                     |
++-----------+------+---------------------------------------------------+
+3 rows in set (0.00 sec)
+```
+
+`USER_ATTRIBUTES` 表中列的含义如下：
+
+* `USER`：用户名。
+* `HOST`：用户的主机名。
+* `ATTRIBUTE`：与当前用户相关的注释和属性。
```
Comment (qiancai) [inline information-schema/information-schema-user-attributes.md:51]:
```suggestion
```

File: sql-statements/sql-statement-alter-user.md [hunk 6]
```diff
@@ -91,6 +93,16 @@ ALTER USER 'newuser' ACCOUNT LOCK;
...
+
+```sql
+ALTER USER 'newuser' ATTRIBUTE '{"newAttr": "value", "deprecatedAttr": null}';
+```
+
+```
+Query OK, 0 rows affected (0.02 sec)
+```
+
 > **注意：**
```
Comment (qiancai) [inline sql-statements/sql-statement-alter-user.md:106]: @CbcWestwolf 这里可以请再加一个修改 comment 的示例语句吗，谢谢

File: sql-statements/sql-statement-alter-user.md [hunk 7]
```diff
@@ -91,6 +93,36 @@ ALTER USER 'newuser' ACCOUNT LOCK;
 Query OK, 0 rows affected (0.02 sec)
 ```
 
+{{< copyable "sql" >}}
+
```
Comment (qiancai) [inline sql-statements/sql-statement-alter-user.md:97]:
```suggestion
```

File: sql-statements/sql-statement-alter-user.md [hunk 8]
```diff
@@ -91,6 +93,36 @@ ALTER USER 'newuser' ACCOUNT LOCK;
...
+```sql
+ALTER USER 'newuser' ATTRIBUTE '{"newAttr": "value", "deprecatedAttr": null}';
+```
+
+```
+Query OK, 0 rows affected (0.02 sec)
+```
+
+{{< copyable "sql" >}}
+
```
Comment (qiancai) [inline sql-statements/sql-statement-alter-user.md:107]:
```suggestion
```

File: sql-statements/sql-statement-alter-user.md [hunk 9]
```diff
@@ -91,6 +93,36 @@ ALTER USER 'newuser' ACCOUNT LOCK;
...
+```sql
+ALTER USER 'newuser' COMMENT 'Here is the comment';
+```
+
+```
+Query OK, 0 rows affected (0.02 sec)
+```
+
+{{< copyable "sql" >}}
+
```
Comment (qiancai) [inline sql-statements/sql-statement-alter-user.md:117]:
```suggestion
```

**Review by qiancai — APPROVED** [review]

File: information-schema/information-schema-user-attributes.md [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+| ATTRIBUTE | longtext     | YES  |      | NULL    |       |
++-----------+--------------+------+------+---------+-------+
+3 rows in set (0.00 sec)
+```
+
+`USER_ATTRIBUTES` 表中列的含义如下：
+
+* `USER`：用户名。
+* `HOST`：用户可用于连接的主机。百分号（`％`）表示主机名不受限制。
+* `ATTRIBUTE`：与当前用户相关的注释和属性。
```
Comment (qiancai) [inline information-schema/information-schema-user-attributes.md:30]:
```suggestion
* `ATTRIBUTE`：通过 CREATE USER 或 ALTER USER 语句设置的当前用户相关的注释和属性。
```

File: information-schema/information-schema.md [hunk 11]
```diff
@@ -43,6 +43,7 @@ Information Schema 提供了一种查看系统元数据的 ANSI 标准方法。
 | [`TABLE_CONSTRAINTS`](/information-schema/information-schema-table-constraints.md)                                         | 提供有关主键、唯一索引和外键的信息。                                              |
 | `TABLE_PRIVILEGES`                                                                                                         | TiDB 未实现，返回零行。                                                           |
 | `TRIGGERS`                                                                                                                 | TiDB 未实现，返回零行。                                                           |
+| [`USER_ATTRIBUTES`](/information-schema/information-schema-user-attributes.md)                                             | 汇总与当前用户相关的注释和属性。                                                        |
```
[the remark this answers is not in the document]
Comment (qiancai) [reply information-schema/information-schema.md:46]: reason: 应该不只是当前用户的？

File: information-schema/information-schema-user-attributes.md [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
++-----------+------+---------------------------------------------------+
+| USER      | HOST | ATTRIBUTE                                         |
++-----------+------+---------------------------------------------------+
+| root      | %    | NULL                                              |
+| testuser1 | %    | {"comment": "This user is created only for test"} |
+| testuser2 | %    | {"email": "user@pingcap.com"}                     |
++-----------+------+---------------------------------------------------+
+3 rows in set (0.00 sec)
+```
+
```
Comment (Oreoxmt) [inline information-schema/information-schema-user-attributes.md:50]:
```suggestion
```

File: sql-statements/sql-statement-alter-user.md [hunk 13]
```diff
@@ -91,6 +93,30 @@ ALTER USER 'newuser' ACCOUNT LOCK;
...
 
+```sql
+ALTER USER 'newuser' ATTRIBUTE '{"newAttr": "value", "deprecatedAttr": null}';
+```
+
+```
+Query OK, 0 rows affected (0.02 sec)
+```
+
+```sql
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-alter-user.md:104]:
````suggestion

通过 `ALTER USER ... COMMENT` 修改用户 `newuser` 的注释：

```sql
````

File: sql-statements/sql-statement-alter-user.md [hunk 14]
```diff
@@ -91,6 +93,30 @@ ALTER USER 'newuser' ACCOUNT LOCK;
...
+ALTER USER 'newuser' COMMENT 'Here is the comment';
+```
+
+```
+Query OK, 0 rows affected (0.02 sec)
+```
+
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-alter-user.md:111]:
```suggestion

通过 `ALTER USER ... ATTRIBUTE` 修改用户 `newuser` 的注释：

```

File: sql-statements/sql-statement-create-user.md [hunk 15]
```diff
@@ -95,6 +97,30 @@ CREATE USER 'newuser5'@'%' ACCOUNT LOCK;
 Query OK, 1 row affected (0.02 sec)
 ```
 
+创建一个带注释的用户。
+
+{{< copyable "sql" >}}
+
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-create-user.md:103]:
```suggestion

```

File: sql-statements/sql-statement-create-user.md [hunk 16]
```diff
@@ -95,6 +97,30 @@ CREATE USER 'newuser5'@'%' ACCOUNT LOCK;
...
+```
+
+```
+Query OK, 1 row affected (0.02 sec)
+```
+
+创建一个具有邮箱 (`email`) 属性的用户。
+
+{{< copyable "sql" >}}
+
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-create-user.md:115]:
```suggestion

```

File: sql-statements/sql-statement-alter-user.md [hunk 17]
```diff
@@ -91,6 +93,30 @@ ALTER USER 'newuser' ACCOUNT LOCK;
 Query OK, 0 rows affected (0.02 sec)
 ```
 
+```sql
+ALTER USER 'newuser' ATTRIBUTE '{"newAttr": "value", "deprecatedAttr": null}';
+```
+
+```
+Query OK, 0 rows affected (0.02 sec)
+```
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-alter-user.md:102]: 建议改成 `SELECT * FROM user_attributes;` 的输出

Comment (Oreoxmt) [inline sql-statements/sql-statement-alter-user.md:102]:
````suggestion
修改 `newuser` 的属性：

```sql
ALTER USER 'newuser' ATTRIBUTE '{"newAttr": "value", "deprecatedAttr": null}';
SELECT * FROM user_attributes;
```

```sql
+-----------+------+---------------------------------------------------+
| USER      | HOST | ATTRIBUTE                                         |
+-----------+------+---------------------------------------------------+
| newuser   | %    | {"newAttr": "value", "deprecatedAttr": null}      |
+-----------+------+---------------------------------------------------+
1 rows in set (0.00 sec)
```
````

File: sql-statements/sql-statement-alter-user.md [hunk 18]
```diff
@@ -91,6 +93,43 @@ ALTER USER 'newuser' ACCOUNT LOCK;
...
+
+通过 `ALTER USER ... COMMENT` 修改用户 `newuser` 的注释：
+
+```sql
+ALTER USER 'newuser' COMMENT 'Here is the comment';
+```
+
+```
+Query OK, 0 rows affected (0.02 sec)
+```
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-alter-user.md:121]:
````suggestion
```sql
ALTER USER 'newuser' COMMENT 'Here is the comment';
SELECT * FROM user_attributes;
```

```sql
+-----------+------+---------------------------------------------------+
| USER      | HOST | ATTRIBUTE                                         |
+-----------+------+---------------------------------------------------+
| newuser   | %    | {"comment": "Here is the comment", "newAttr": "value", "deprecatedAttr": null}      |
+-----------+------+---------------------------------------------------+
1 rows in set (0.00 sec)
```
````

File: sql-statements/sql-statement-alter-user.md [hunk 19]
```diff
@@ -91,6 +93,43 @@ ALTER USER 'newuser' ACCOUNT LOCK;
...
+
+通过 `ALTER USER ... ATTRIBUTE` 修改用户 `newuser` 的注释：
+
+```sql
+ALTER USER 'newuser' ATTRIBUTE '{"comment": null}';
+```
+
+```
+Query OK, 0 rows affected (0.02 sec)
+```
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-alter-user.md:131]:
````suggestion
```sql
ALTER USER 'newuser' ATTRIBUTE '{"comment": null}';
SELECT * FROM user_attributes;
```

```sql
+-----------+------+---------------------------------------------------+
| USER      | HOST | ATTRIBUTE                                         |
+-----------+------+---------------------------------------------------+
| newuser   | %    | {"comment": null, "newAttr": "value", "deprecatedAttr": null}      |
+-----------+------+---------------------------------------------------+
1 rows in set (0.00 sec)
```
````

File: sql-statements/sql-statement-create-user.md [hunk 20]
```diff
@@ -95,6 +97,24 @@ CREATE USER 'newuser5'@'%' ACCOUNT LOCK;
...
 ```
 
+创建一个带注释的用户。
+```sql
+CREATE USER 'newuser6'@'%' COMMENT 'This user is created only for test';
+```
+
+```
+Query OK, 1 row affected (0.02 sec)
+```
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-create-user.md:107]:
````suggestion
创建一个带注释的用户。

```sql
CREATE USER 'newuser6'@'%' COMMENT 'This user is created only for test';
SELECT * FROM user_attributes;
```

```
+-----------+------+---------------------------------------------------+
| USER      | HOST | ATTRIBUTE                                         |
+-----------+------+---------------------------------------------------+
| newuser6  | %    | {"comment": "This user is created only for test"} |
+-----------+------+---------------------------------------------------+
1 rows in set (0.00 sec)
```
````

File: sql-statements/sql-statement-create-user.md [hunk 21]
```diff
@@ -95,6 +97,24 @@ CREATE USER 'newuser5'@'%' ACCOUNT LOCK;
...
+
+创建一个具有邮箱 (`email`) 属性的用户。
+```sql
+CREATE USER 'newuser7'@'%' ATTRIBUTE '{"email": "user@pingcap.com"}';
+```
+
+```
+Query OK, 1 row affected (0.02 sec)
+```
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-create-user.md:116]:
````suggestion
创建一个具有邮箱 (`email`) 属性的用户。

```sql
CREATE USER 'newuser7'@'%' ATTRIBUTE '{"email": "user@pingcap.com"}';
SELECT * FROM user_attributes;
```

```sql
+-----------+------+---------------------------------------------------+
| USER      | HOST | ATTRIBUTE                                         |
+-----------+------+---------------------------------------------------+
| newuser7  | %    | {"email": "user@pingcap.com"} |
+-----------+------+---------------------------------------------------+
1 rows in set (0.00 sec)
```
````

**Review by Oreoxmt — APPROVED** [review]

**Comment by CbcWestwolf** [comment]
attribute 里面 `{"key": null}` 是删除 `key` 哈，不会出现在查询结果中

File: sql-statements/sql-statement-alter-user.md [hunk 22]
```diff
@@ -91,6 +93,55 @@ ALTER USER 'newuser' ACCOUNT LOCK;
...
+```sql
++-----------+------+--------------------------+
+| USER      | HOST | ATTRIBUTE                |
++-----------+------+--------------------------+
+| newuser   | %    | {"newAttr": "value"}     |
++-----------+------+--------------------------+
+1 rows in set (0.00 sec)
+```
+
+
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-alter-user.md:112]:
```suggestion
```

File: sql-statements/sql-statement-alter-user.md [hunk 23]
```diff
@@ -91,6 +93,54 @@ ALTER USER 'newuser' ACCOUNT LOCK;
...
+```sql
++-----------+------+---------------------------+
+| USER      | HOST | ATTRIBUTE                 |
++-----------+------+---------------------------+
+| newuser   | %    | {"newAttr": "value"}      |
++-----------+------+---------------------------+
+1 rows in set (0.00 sec)
+```
+
 > **注意：**
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-alter-user.md:144]: L150 ``* 在 MySQL 中，`ALTER` 语句用于更改属性，例如使密码失效。但 TiDB 尚不支持此功能。`` 需要修改吗

**Review by Oreoxmt — APPROVED** [review]
LGTM

File: information-schema/information-schema-user-attributes.md [hunk 24]
```diff
@@ -0,0 +1,49 @@
...
+* `USER`：用户名。
+* `HOST`：用户可用于连接的主机。百分号（`％`）表示主机名不受限制。
+* `ATTRIBUTE`：通过 [`CREATE USER`](/sql-statements/sql-statement-create-user.md) 或 [`ALTER USER`](/sql-statements/sql-statement-alter-user.md) 语句设置的用户相关的注释和属性。
+
+示例：
+
+```sql
+CREATE USER testuser1 COMMENT 'This user is created only for test';
+CREATE USER testuser2 ATTRIBUTE '{"email": "user@pingcap.com"}';
+SELECT * FROM user_attributes;
```
Comment (qiancai) [inline information-schema/information-schema-user-attributes.md:37]:
```suggestion
SELECT * FROM information_schema.user_attributes;
```

**Pull request merged**